### PR TITLE
SONAR-10411 Dot not persist anymore name of provided links

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java [hunk 1]
```diff
@@ -64,27 +64,26 @@ public void handle(Request request, Response response) throws Exception {
...
+  private static void checkNotProvided(ProjectLinkDto link) {
     String type = link.getType();
     boolean isProvided = type != null && PROVIDED_TYPES.contains(type);
     WsUtils.checkRequest(!isProvided, "Provided link cannot be deleted.");
   }
 
-  private void checkProjectAdminPermission(ComponentLinkDto link) {
-    userSession.checkComponentUuidPermission(UserRole.ADMIN, link.getComponentUuid());
+  private void checkProjectAdminPermission(ProjectLinkDto link) {
+    userSession.checkComponentUuidPermission(UserRole.ADMIN, link.getProjectUuid());
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java]: Code Smell Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGKOxDTuvYJzwp2ZAje&open=AWGKOxDTuvYJzwp2ZAje)

[ISSUE_KEY]: AWGKOxDTuvYJzwp2ZAje

Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java]: Code Smell Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGKOxDTuvYJzwp2ZAje&open=AWGKOxDTuvYJzwp2ZAje)

[ISSUE_KEY]: AWGKOxDTuvYJzwp2ZAje

Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java]: Code Smell Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGKOxDTuvYJzwp2ZAje&open=AWGKOxDTuvYJzwp2ZAje)

[ISSUE_KEY]: AWGKOxDTuvYJzwp2ZAje

Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java]: Code Smell Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGKOxDTuvYJzwp2ZAje&open=AWGKOxDTuvYJzwp2ZAje)

[ISSUE_KEY]: AWGKOxDTuvYJzwp2ZAje

Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java]: Code Smell Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGKOxDTuvYJzwp2ZAje&open=AWGKOxDTuvYJzwp2ZAje)

[ISSUE_KEY]: AWGKOxDTuvYJzwp2ZAje

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java [hunk 2]
```diff
@@ -328,7 +328,9 @@
 
   /**
    * @since 3.3
+   * @deprecated since 7.1, developer connection link is no more feed
    */
+  @Deprecated
   String LINKS_SOURCES_DEV = "sonar.links.scm_dev";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: Code Smell Code Smell: Do not forget to remove this deprecated code someday. ([squid:S1133](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGK4YyXuvYJzwp2ZGjQ&open=AWGK4YyXuvYJzwp2ZGjQ)

[ISSUE_KEY]: AWGK4YyXuvYJzwp2ZGjQ

File: server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java (hunk 1 above)
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java]: Code Smell Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGKOxDTuvYJzwp2ZAje&open=AWGKOxDTuvYJzwp2ZAje)

[ISSUE_KEY]: AWGKOxDTuvYJzwp2ZAje

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java (hunk 2 above)
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: Code Smell Code Smell: Do not forget to remove this deprecated code someday. ([squid:S1133](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGK4YyXuvYJzwp2ZGjQ&open=AWGK4YyXuvYJzwp2ZGjQ)

[ISSUE_KEY]: AWGK4YyXuvYJzwp2ZGjQ

File: server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java (hunk 1 above)
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java]: Code Smell Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGKOxDTuvYJzwp2ZAje&open=AWGKOxDTuvYJzwp2ZAje)

[ISSUE_KEY]: AWGKOxDTuvYJzwp2ZAje

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java (hunk 2 above)
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: Code Smell Code Smell: Do not forget to remove this deprecated code someday. ([squid:S1133](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGK4YyXuvYJzwp2ZGjQ&open=AWGK4YyXuvYJzwp2ZGjQ)

[ISSUE_KEY]: AWGK4YyXuvYJzwp2ZGjQ

File: server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java (hunk 1 above)
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java]: Code Smell Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGKOxDTuvYJzwp2ZAje&open=AWGKOxDTuvYJzwp2ZAje)

[ISSUE_KEY]: AWGKOxDTuvYJzwp2ZAje

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistProjectLinksStep.java [hunk 3]
```diff
@@ -44,12 +42,11 @@
 import static org.sonar.server.computation.task.projectanalysis.component.ComponentVisitor.Order.PRE_ORDER;
 
 /**
- * Persist project and module links
+ * Persist project links
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistProjectLinksStep.java]: You could even drop this comment :-)

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistProjectLinksStep.java [hunk 4]
```diff
@@ -89,17 +85,8 @@ private ProjectLinkVisitor(DbSession session) {
 
     @Override
     public void visitProject(Component project) {
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistProjectLinksStep.java]: I may be wrong, but using the visitor pattern (DepthTraversalTypeAwareCrawler and TypeAwareVisitorAdapter) is overkill now that only project is relevant. I suggest to simply call use `treeRootHolder.getRoot()` to process links.

File: server/sonar-db-dao/src/main/java/org/sonar/db/DbClient.java [hunk 5]
```diff
@@ -316,8 +316,8 @@ public FileSourceDao fileSourceDao() {
     return fileSourceDao;
   }
 
-  public ComponentLinkDao componentLinkDao() {
-    return componentLinkDao;
+  public ProjectLinkDao componentLinkDao() {
```
Comment (simonbrandhof) [inline server/sonar-db-dao/src/main/java/org/sonar/db/DbClient.java]: method should be renamed `projectLinkDao()`. Note that it impacts core plugins.

File: server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java [hunk 6]
```diff
@@ -64,27 +64,26 @@ public void handle(Request request, Response response) throws Exception {
...
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/DeleteAction.java]: Code Smell Code Smell: Remove this use of "checkComponentUuidPermission"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGKOxDTuvYJzwp2ZAje&open=AWGKOxDTuvYJzwp2ZAje)

[ISSUE_KEY]: AWGKOxDTuvYJzwp2ZAje

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java (hunk 2 above)
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: Code Smell Code Smell: Do not forget to remove this deprecated code someday. ([squid:S1133](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10411%2Fremove_project_links_name_in_ce&issues=AWGK4YyXuvYJzwp2ZGjQ&open=AWGK4YyXuvYJzwp2ZGjQ)

[ISSUE_KEY]: AWGK4YyXuvYJzwp2ZGjQ

**Pull request merged**